duckduckgo/iOS
# Add delegated autofill isEnabled check and pixel params

Task/Issue URL:
Tech Design URL:
CC:

**Description**:
This was originally merged along with the app group change, but the app group change had an issue so was reverted. This just adds back the none problematic things:

- Adds two additional pixels to see if the check for secure vault being available is being spammed
- Adds param to see if the app is backgrounded, this will help us get more certainty on the secure vault errors
- Adds a param if the user is internal, so in future if an error pixel is only firing for internal users we can calm down



**Steps to test this PR**:
1. The most important thing to test is that autofill still works. See here for autofill testing steps: https://github.com/duckduckgo/iOS/pull/1184 . Also test that the app doesn't crash when in the background for >10 seconds, since that's the issue the original PR introduced
2. Check that all pixels have the is internal user flag. 
3. In FeatureFlagger, hard code internal user to false, pixels should not have the param at all (not even set to false) 
4. When you load a page check that "m_secure-vault_is-enabled-checked_when-enabled" and "m_secure-vault_is-enabled-checked_when-disabled" are being sent appropriately (hard code it off in FeatureFlagger to check the latter). Check they also have the isBackgrounded flag
5. See https://github.com/duckduckgo/iOS/pull/1212 to test secure vault error pixels, check they have the isBackgrounded param



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
